src/services/codefixes/importFixes.ts

```diff
@@ -620,4 +666,73 @@ namespace ts.codefix {
         // Need `|| "_"` to ensure result isn't empty.
         return !isStringANonContextualKeyword(res) ? res || "_" : `_${res}`;
     }
+
+    function createLazyPackageJsonDependencyReader(fromFile: SourceFile, host: LanguageServiceHost) {
+        const packageJsonPaths = findPackageJsons(getDirectoryPath(fromFile.fileName), host);
+        const dependencyIterator = readPackageJsonDependencies();
+        let seenDeps: Map<true> | undefined;
+        function *readPackageJsonDependencies() {
+            type PackageJson = Record<typeof dependencyKeys[number], Record<string, string> | undefined>;
+            const dependencyKeys = ["dependencies", "devDependencies", "optionalDependencies"] as const;
+            for (const fileName of packageJsonPaths) {
+                const content = readJson(fileName, { readFile: host.readFile ? host.readFile.bind(host) : sys.readFile }) as PackageJson;
+                for (const key of dependencyKeys) {
+                    const dependencyHash = content[key];
+                    if (!dependencyHash) {
+                        continue;
+                    }
+                    for (const packageName in dependencyHash) {
+                        yield packageName;
+                    }
+                }
+            }
+        }
+
+        function containsDependency(dependency: string) {
+            if ((seenDeps || (seenDeps = createMap())).has(dependency)) {
+                return true;
+            }
+            let packageName: string;
+            while (packageName = dependencyIterator.next().value) {
+                seenDeps.set(packageName, true);
+                if (packageName === dependency) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        let usesNodeCoreModules: boolean | undefined;
+        function allowsImporting(moduleSpecifier: string): boolean {
+            if (!packageJsonPaths.length) {
+                return true;
+            }
+
+            // If we’re in JavaScript, it can be difficult to tell whether the user wants to import
+            // from Node core modules or not. We can start by seeing if the user is actually using
+            // any node core modules, as opposed to simply having @types/node accidentally as a
+            // dependency of a dependency.
+            if (isSourceFileJS(fromFile) && JsTyping.nodeCoreModules.has(moduleSpecifier)) {
+                if (usesNodeCoreModules === undefined) {
+                    usesNodeCoreModules = consumesNodeCoreModules(fromFile);
+                }
+                if (usesNodeCoreModules) {
+                    return true;
+                }
+            }
+
+            return containsDependency(moduleSpecifier)
+                || containsDependency(getTypesPackageName(moduleSpecifier));
+        }
+
+        return {
+            packageJsonPaths,
+            containsDependency,
+            allowsImporting

```

Comment:
it looks like only allowsImporting is used, so it should be possible to drop the other two from the returned value.